snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedDescriptionAcceptabilityUpdateRequest.java

```diff
@@ -59,49 +63,95 @@ protected boolean canUpdate(TransactionContext context) {
 					context.update(oldRevision, updatedMember.build());
-				}
 
-				// Remove it from the working list, as we have found a match
-				newLanguageMembersToCreate.remove(languageReferenceSetId);
-			} else if (newLanguageMembersToCreate.containsKey(languageReferenceSetId)) {
-				// Change acceptability, set status to active if required, place it in the supplied module
-				final Acceptability newAcceptability = newLanguageMembersToCreate.get(languageReferenceSetId);
-				updatedMember.field(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID, newAcceptability.getConceptId());
-				ensureMemberActive(context, existingMember, updatedMember);
-				updateModule(context, existingMember, updatedMember, moduleIdSupplier.apply(componentToUpdate));
-				unsetEffectiveTime(existingMember, updatedMember);
-				context.update(oldRevision, updatedMember.build());
-				newLanguageMembersToCreate.remove(languageReferenceSetId);
+				} else {
+					// Case 1.2: There is at least one match for acceptability
+					final List<SnomedReferenceSetMember> activeMatches = acceptabilityMatches.stream()
+							.filter(m -> Objects.equal(newAcceptability.getConceptId(), m.getProperties().get(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID)))
+							.collect(Collectors.toList());
+					
+					if (activeMatches.isEmpty()) {
+						// Case 1.2.1: There is no active match, reactivate one of the members, preferably an extension one
+						existingMember = select(acceptabilityMatches, moduleId, Concepts.MODULE_SCT_CORE);
+						final SnomedRefSetMemberIndexEntry.Builder updatedMember = SnomedRefSetMemberIndexEntry.builder(existingMember);
+						final SnomedRefSetMemberIndexEntry oldRevision = updatedMember.build();
+						
+						if (ensureMemberActive(context, existingMember, updatedMember)) {
+							context.update(oldRevision, updatedMember.build());
+						}
+					} else {
+						// Case 1.2.2: There is at least one active match, keep the INT one if available
+						existingMember = select(activeMatches, Concepts.MODULE_SCT_CORE, moduleId);
+					}
+				}
+					
+				// Remove/deactivate any remaining members in the same language reference set
+				for (SnomedReferenceSetMember otherMember : refsetMatches) {
+					
+					if (existingMember == null || !otherMember.getId().equals(existingMember.getId())) {
+						final SnomedRefSetMemberIndexEntry.Builder updatedMember = SnomedRefSetMemberIndexEntry.builder(otherMember);
+						final SnomedRefSetMemberIndexEntry oldRevision = updatedMember.build();
+						
+						if (removeOrDeactivate(context, otherMember, updatedMember)) {
+							context.update(oldRevision, updatedMember.build());
+						}
+					}
+				}
+				
 			} else {
-				// Not acceptable in this language reference set, remove or inactivate if already released
-				if (removeOrDeactivate(context, existingMember, updatedMember)) {
+				// Case 2: No match exists for language reference set
+				newLanguageMembersToCreate.put(languageReferenceSet, newAcceptability);
```

Comment:
`newLanguageMembersToCreate` are now collected but don't seem to be used... :perplexed:

Reply:
They have always been collected, and are used.

Reply:
Oops, missed the unchanged block of code at the end of the file, sorry about that.